pipeline/pipeline_runner.go

```diff
@@ -158,7 +158,7 @@ type PipelinePack struct {
 // Returns a new PipelinePack pointer that will recycle itself onto the
 // provided channel when a message has completed processing.
 func NewPipelinePack(recycleChan chan *PipelinePack) (pack *PipelinePack) {
-	msgBytes := make([]byte, message.MAX_MESSAGE_SIZE)
+	msgBytes := make([]byte, 0, message.MAX_MESSAGE_SIZE)
```

Comment:
One [place](https://github.com/mozilla-services/heka/blob/dev/pipeline/pipeline_runner.go#L215) where it happens is my fault, it just landed. I'm pretty sure I tried to make this change earlier and hit a side-effect somewhere else, too, but I just searched through the code and couldn't find anything.

I'm fine w/ trying this out, but the code I linked to above should be changed so the resize happens before the copy, and the Zero method should be changed to match the new pack behavior.